libs/gpg45-evaluator/src/main/java/uk/gov/di/ipv/core/library/gpg45/Gpg45ProfileEvaluator.java

```diff
@@ -63,6 +67,91 @@ public Gpg45Scores buildScore(List<VerifiableCredential> vcs) {
                 .build();
     }
 
+    private List<VerifiableCredential> deduplicateVcsByDocumentType(
+            List<VerifiableCredential> vcs) {
+        var result = new ArrayList<VerifiableCredential>();
+        var deduplicatedVcs = new HashMap<String, VerifiableCredential>();
+        for (var vc : vcs) {
+            var docType = getVcDocumentType(vc);
+            if (isEmpty(docType)) {
+                result.add(vc);
+            } else {
+                var existing = deduplicatedVcs.get(docType);
+                if (existing == null) {
+                    deduplicatedVcs.put(docType, vc);
+                } else {
+                    deduplicatedVcs.put(
+                            docType, selectVerifiableCredentialWithSameDocumentType(existing, vc));
+                }
```

Comment:
If you manage to boil this down to Evidence (rather than VCs), you could streamline this a bit:
```
var existing = deduplicatedEvidence.get(docType);
if (existing == null || vc.compareTo(existing) > 0) {
  deduplicatedEvidence.put(docType, vc);
}
```
where you make `Evidence` implement the `Comparable` interface